app/services/file_appender.rb

```diff
@@ -202,5 +202,18 @@ def adjust_pending_uploads(resource)
       return unless resource.respond_to?(:pending_uploads)
       resource.pending_uploads = [] if resource.pending_uploads.blank?
       resource.pending_uploads = (resource.pending_uploads || []) - files
+
+      # This is needed for newly-created PendingUploads for uploads to the same UploadFile
```

Comment:
I'm confused. What's an `UploadFile` and why am I uploading into it?

Reply:
we think we can get rid of this block now, due to improved equality logic in valkyrie